pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -126,14 +126,6 @@ func (pg *pgcontroller) createNormalPodPGIfNotExist(pod *v1.Pod) error {
 }
 
 func newPGOwnerReferences(pod *v1.Pod) []metav1.OwnerReference {
-	if len(pod.OwnerReferences) != 0 {
```

Comment:
why remove this?

Reply:
> why remove this?

+1

Reply:
The function `newPGOwnerReferences` returns what to set as the owner of the podgroup that is being created.  Without the modifications above, what it says is if the pod is independent (it is created using something like `kubectl apply -f pod.yaml`) the owner of the podgroup is set to the pod.  Because `blockOwnerDeletion=true` it will need to modify that pod's finalizers.  These are the permissions being added in this PR.  If the pod is owned by a controller, such as if a replicaset is creating the pods, this block of code will try to set the owner of the podgroup being created to be the owner of the pod.  If a replicaset is creating a pod, this code will try to set the owner of the podgroup to be the replicaset that created that pod.  Because `blockOwnerDeletion=true`, setting the owner of the podgroup to the replicaset will require a modification to the replicaset finalizers.

If we remove that block of code volcano controller will only require RBACs for pod/finalizers when it is scheduling independent pods because the owners of the podgroups created though this code path will only ever be pods.  It will not require permissions for finalizers for other resources.

In our cluster we want to use volcano to schedule the pods for several workloads.  Currently we are using volcano to schedule vcjobs, but we want to expand it to schedule pods for other workflows.  As we add additional workloads with additional CRDs it would be difficult to continue to add additional permissions for the finalizers for each new CRD.

I am not sure if this explanation helped.  As I have tested using volcano to schedule arbitrary pods in a cluster that enforces RBACs, removing this portion of code made it so pods would get scheduled.  Otherwise, many pods would never be scheduled because the podgroup was never created.

Reply:
> I am not sure if this explanation helped. As I have tested using volcano to schedule arbitrary pods in a cluster that enforces RBACs, removing this portion of code made it so pods would get scheduled. Otherwise, many pods would never be scheduled because the podgroup was never created.

I'd suggest to only commit RBAC part in this pr, and share more logs for this case.  We'd like to support all kind of workloads in one cluster, so it's important to clarify the root cause :)

Reply:
Sounds good.  I looked into this a little more.  I think the reason why I am seeing this issue and other have not might be because of changes to out Kubernetes is configured.  In the cluster where I am seeing this issue, we enable [OwnerReferencesPermissionEnforcement](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#ownerreferencespermissionenforcement).  According to the docs, `/update` for the finalizer will be required to set `blockOwnerDeletion`.  However, I can take a little more time to explore options for resolving this issue on our cluster.

Reply:
please feel free to share the log and discuss it in the community :)